### PR TITLE
Update mongodb usecase

### LINKED ISSUE #822: Create MongoDB usecase for 0.5
## Is this a BUG REPORT or FEATURE REQUEST?
 
 Choose one: BUG REPORT or FEATURE REQUEST

**What happened**:

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by mahebbar**
**What this PR does / why we need it**: New feature for 0.5

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #822

### CODE REVIEW

File: documentation/source/Usecases/mongodb.rst [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+  ::
+
+    sudo apt-get install <packagename>:
+
+    make
+    libsas12-dev
+    libssl-dev
+    libmongoc-dev
+    libbson-dev
+
```
Comment (ksatchit) [inline documentation/source/Usecases/mongodb.rst]: We need to include the following packages also to the above list : 

- automake
- libmysqlclient-dev
- libtool

File: documentation/source/Usecases/mongodb.rst [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+    openebs-standard     openebs.io/provisioner-iscsi
+    openebs-zk           openebs.io/provisioner-iscsi
+
+2. Deploy the mongo-statefulset with OpenEBS storage
+------------------------------------------------------
+
+Use OpenEBS as persistent storage for the MongoDB statefulset by selecting an OpenEBS storage class in the persistent volume claim. A sample MongoDB statefulset yaml (with container attributes and pvc details) is available in the OpenEBS git repository.
+
+The number of replicas in the statefulset can be modified as required. The following example uses two replicas. The replica count can be edited in the statefulset specification.
+::
```
Comment (ksatchit) [inline documentation/source/Usecases/mongodb.rst]: The number of replicas in the statefulset can be modified as required. The following example uses three replicas

File: documentation/source/Usecases/mongodb.rst [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+  
+  ---
+  apiVersion: apps/v1beta1
+  kind: StatefulSet
+  metadata:
+   name: mongo
+  spec:
+   serviceName: "mongo"
+   replicas: 2
```
Comment (ksatchit) [inline documentation/source/Usecases/mongodb.rst]: replicas: 3

File: documentation/source/Usecases/mongodb.rst [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+    root@maya-apiserver-1089964587-x5q15:/#
+
+Obtain the list of OpenEBS persistent volumes created by the MongoDB statefulset application YAML.
+::
+
+    root@maya-apiserver-1089964587-x5q15:/# maya volume list
+    Name                                      Status
+    pvc-0d39583c-bad7-11e7-869d-000c298ff5fc  Running
+    pvc-21da76b6-bad7-11e7-869d-000c298ff5fc  Running
+
```
Comment (ksatchit) [inline documentation/source/Usecases/mongodb.rst]:
```
     root@maya-apiserver-1089964587-x5q15:/# maya volume list
     Name                                      Status
     pvc-0d39583c-bad7-11e7-869d-000c298ff5fc  Running
     pvc-21da76b6-bad7-11e7-869d-000c298ff5fc  Running
      :  
      ```

**Comment by ksatchit** [comment]
@mahebbar , the mongodb documentation looks good. There are a few changes in the content which I discovered some tests recently and would like this to be added - have added comments at the respective location. 

I need to make these changes in the developer docs as well

File: documentation/source/Usecases/mongodb.rst (hunk 1 above)
Comment (mahebbar) [reply documentation/source/Usecases/mongodb.rst]: Done!

File: documentation/source/Usecases/mongodb.rst (hunk 2 above)
Comment (mahebbar) [reply documentation/source/Usecases/mongodb.rst]: Done! Thought as much :)

File: documentation/source/Usecases/mongodb.rst (hunk 3 above)
Comment (mahebbar) [reply documentation/source/Usecases/mongodb.rst]: Done!

File: documentation/source/Usecases/mongodb.rst (hunk 4 above)
Comment (mahebbar) [reply documentation/source/Usecases/mongodb.rst]: Done!

Comment (mahebbar) [reply documentation/source/Usecases/mongodb.rst]: There is a sentence "Take an interactive bash session into the maya-apiserver pod container."
Can we say "include an interactive..." or "run an interactive..."?

**Pull request merged**